### PR TITLE
Decorate the disassembly with GCInfo

### PR DESCRIPTION
**Pull request by eduardo-vp**
Fixes https://github.com/icsharpcode/ILSpy/issues/1885

Shows GC Info at the beginning and the GC transitions as well. Also it stops iterating through the whole `runtimeFunction.DebugInfo.BoundsList` for each instruction for a slight performance improvement.

image

### CODE REVIEW

File: ILSpy.ReadyToRun/Properties/Resources.zh-Hans.resx [hunk 1]
```diff
@@ -129,4 +129,7 @@
   <data name="ShowUnwindInfo" xml:space="preserve">
     <value>显示展开信息</value>
   </data>
+  <data name="ShowGCInfo" xml:space="preserve">
+    <value>?</value>
```
Comment (eduardo-vp) [inline ILSpy.ReadyToRun/Properties/Resources.zh-Hans.resx:133]: Don't know how to get a proper value here

File: ILSpy.ReadyToRun/ReadyToRunDisassembler.cs [hunk 2]
```diff
@@ -47,10 +47,19 @@ public ReadyToRunDisassembler(ITextOutput output, ReadyToRunReader reader, Runti
 
 		public void Disassemble(PEFile currentFile, int bitness, ulong address, bool showMetadataTokens, bool showMetadataTokensInBase10)
 		{
-			// TODO: Decorate the disassembly with GCInfo
 			ReadyToRunMethod readyToRunMethod = runtimeFunction.Method;
 			WriteCommentLine(readyToRunMethod.SignatureString);
 
+			if (ReadyToRunOptions.GetIsShowGCInfo(null) && readyToRunMethod.GcInfo != null)
```
Comment (eduardo-vp) [inline ILSpy.ReadyToRun/ReadyToRunDisassembler.cs:53]: Also wondering if it's possible that `readyToRunMethod.GcInfo` returns null, in that case should we show a comment that the GC Info was not found (if the checkbox was enabled)?

File: ILSpy.ReadyToRun/Properties/Resources.zh-Hans.resx (hunk 1 above)
Comment (dgrunwald) [reply ILSpy.ReadyToRun/Properties/Resources.zh-Hans.resx:133]: Initially you can leave the chinese version unmodified, it'll fall back to the english resources for missing strings.

**Review by cshung — APPROVED** [review]
LGTM

**Comment by siegfriedpammer** [comment]
Thank you for the contribution!

**Pull request merged**